httpbakery/keyring.go

```diff
@@ -0,0 +1,97 @@
+package httpbakery
+
+import (
+	"encoding/json"
+	"io/ioutil"
+	"net/http"
+	"net/url"
+	"strings"
+
+	"gopkg.in/errgo.v1"
+
+	"gopkg.in/macaroon-bakery.v1/bakery"
+)
+
+// NewPublicKeyRing returns a new public keyring that uses
+// the given client to find public keys and uses the
+// given cache as a backing. If cache is nil, a new
+// cache will be created. If client is nil, http.DefaultClient will
+// be used.
+func NewPublicKeyRing(client *http.Client, cache *bakery.PublicKeyRing) *PublicKeyRing {
+	if cache == nil {
+		cache = bakery.NewPublicKeyRing()
+	}
+	if client == nil {
+		client = http.DefaultClient
+	}
+	return &PublicKeyRing{
+		client: client,
+		cache:  cache,
```

Comment:
I think leaving it as omitted is reasonable. The sense of the boolean is that way for a reason.